swarm.go

```diff
@@ -0,0 +1,175 @@
+package httpapi
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/interface-go-ipfs-core"
+	inet "github.com/libp2p/go-libp2p-net"
+	"github.com/libp2p/go-libp2p-peer"
+	"github.com/libp2p/go-libp2p-peerstore"
+	"github.com/libp2p/go-libp2p-protocol"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type SwarmAPI HttpApi
+
+func (api *SwarmAPI) Connect(ctx context.Context, pi peerstore.PeerInfo) error {
+	saddrs := make([]string, len(pi.Addrs))
+	for i, addr := range pi.Addrs {
+		saddrs[i] = addr.String()
```

Comment:
I believe these need to end with the peer ID, right?

Reply:
Pushed a fix, not sure if 100% correct (swarm is a bit tricky to test in coreapi)